### DISCUSSION TITLE
[Chapter 08] 읽기만 하는 경우 트랜잭션

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#113] [Chapter 08. Data Persistence] [answered]
select만 하는 경우 `@Transactional`을 굳이 달 필요없는데
`@Transactional(readOnly = true)` 이런 어노테이션을 다는 경우가 있더라구요, 

달지 않은경우와, `@Transactional(readOnly = true)` 차이가 있을까요? 굳이 사용하는 이유가 있는것 같은데..

**Comment by ahngilwoong** [comment, marked as answer]
이번 기회에 왜 쓰는 것인지 확실히 알아 갈 수 있는 시간이었습니다

1. readOnly = true로 설정시 더티체킹 등 필요하지 않는 동작이 줄어들어 성능상의 이점을 가져갈 수 있습니다.
2. 이 코드는 읽기로만 쓰인다! 라고 명시적으로 알아 볼 수 있습니다.
3. readOnly = true로 설정 후 Data 수정시 JPA 프로바이더나 트랜잭션 관리자가 예외를 발생시켜 실수를 원천 차단할 수 있습니다. (해당 내용은 db별로 상이합니다, 데이터 수정이 되는 DB가 있습니다.)

또한 DBMS별로 READ ONLY에 대한 설정 동작 방식이 다르다는 것을 알게 되었습니다.
[DB별 READ ONLY 동작 방식](https://lob-dev.tistory.com/entry/DBMS-%EB%B3%84-Transaction-Read-Only%EC%97%90-%EB%8C%80%ED%95%9C-%EB%8F%99%EC%9E%91-%EB%B0%A9%EC%8B%9D-1)
